packages/bot-skeleton/src/scratch/blocks/Binary/Trade Definition/trade_definition_tradeoptions.js

```diff
@@ -68,6 +69,20 @@ Blockly.Blocks.trade_definition_tradeoptions = {
         };
     },
     onchange(event) {
+        if (event.type === 'change') {
+            const selected_block = this.workspace.getBlockById(event.blockId);
+            selected_block?.parentBlock_?.inputList.filter(
+                item => ['DURATION', 'AMOUNT'].includes(item.name)
+            ).forEach(input => {
+                const input_target = input.connection.targetBlock();
+                const value = input_target.getFieldValue('NUM');
+                if (isFinite(value) && value.startsWith('0')) {
+                    const new_value = value.replace(/^0+/, '');
```

Comment:
why do we need + here?

Reply:
Because users may enter values like `0000x`. We need to remove any number of starting zeros.